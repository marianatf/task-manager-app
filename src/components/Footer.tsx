import React from "react";

interface Props {
  activeCount: number;
  completedCount: number;
  onClearCompleted: () => void;
}

export const Footer: React.FC<Props> = ({
  activeCount = 0,
  completedCount = 0,
  onClearCompleted,
}) => {
  const isSingleActiveTask = activeCount === 1;
  const activeTaskLabel = isSingleActiveTask ? "task" : "tasks";

  return (
    <footer className="flex justify-between items-center px-4 py-2 border-t border-gray-200 text-gray-600 w-full">
      <span>
        <strong className="font-semibold">{activeCount}</strong> outstanding {activeTaskLabel}
      </span>
      {completedCount > 0 && (
        <button
          className="hover:text-red-600 transition"
          onClick={onClearCompleted}
        >
          Clear completed
        </button>
      )}
    </footer>
  );
};
