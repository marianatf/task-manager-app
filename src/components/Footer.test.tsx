import { render, screen, fireEvent } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import { Footer } from "./Footer";

describe("Footer", () => {
  const mockOnClearCompleted = vi.fn();

  it("should match snapshot", () => {
    const { asFragment } = render(
      <Footer
        activeCount={0}
        completedCount={0}
        onClearCompleted={mockOnClearCompleted}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should call onClearCompleted when clicked", () => {
    render(
      <Footer
        activeCount={1}
        completedCount={1}
        onClearCompleted={mockOnClearCompleted}
      />
    );
    fireEvent.click(screen.getByText(/clear completed/i));
    expect(mockOnClearCompleted).toHaveBeenCalled();
  });
});
