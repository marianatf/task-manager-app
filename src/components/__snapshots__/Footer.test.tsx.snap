// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="flex justify-between items-center px-4 py-2 border-t border-gray-200 text-gray-600 w-full"
  >
    <span>
      <strong
        class="font-semibold"
      >
        0
      </strong>
       outstanding tasks
    </span>
  </footer>
</DocumentFragment>
`;
